### PR TITLE
Reduce allocs in AddPeer

### PR DESCRIPTION
**Pull request by aarshkshah1992**
@Stebalien 
For https://github.com/libp2p/go-libp2p-kad-dht/issues/594.

### CODE REVIEW

File: table.go [hunk 1]
```diff
@@ -152,16 +152,16 @@ func (rt *RoutingTable) addPeer(p peer.ID, queryPeer bool) (bool, error) {
...
-	for _, pc := range allPeers {
-		if time.Since(pc.LastUsefulAt) > rt.usefulnessGracePeriod {
-			// let's evict it and add the new peer
-			if bucket.remove(pc.Id) {
-				bucket.pushFront(&PeerInfo{Id: p, LastUsefulAt: lastUsefulAt, LastSuccessfulOutboundQueryAt: time.Now(),
-					dhtId: ConvertPeerID(p)})
-				rt.PeerAdded(p)
-				return true, nil
-			}
+	pid := bucket.selectFirst(func(pi *PeerInfo) bool {
```
Comment (Stebalien) [inline table.go:155]: ot nit: shouldn't we pick the _least_ useful peer?

Comment (aarshkshah1992) [reply table.go:155]: @Stebalien What's the meaning of "ot nit" ?

**Comment by aarshkshah1992** [comment]
@Stebalien While I'm not really sure how much of a change it will make, I've changed the code to use the least useful peer as it sounds like the right thing to do. Please take a look when you can !

File: table.go (hunk 1 above)
Comment (Stebalien) [reply table.go]: Sorry, "off topic nit". That is, a small suggestion that isn't even related to the current patch (just something I noticed while reviewing it).

(I'm basically saying "feel free to ignore this for now, but we should fix this in a future patch").

**Pull request merged**